src/prefect/cli/root.py

```diff
@@ -268,9 +274,20 @@ async def deploy(
 
     Should be run from a project root directory.
     """
-    # TODO: add error handling
-    with open("deployment.yaml", "r") as f:
-        base_deploy = yaml.safe_load(f)
+    if len([value for value in (cron, rrule, interval) if value is not None]) > 1:
+        exit_with_error("Only one schedule type can be provided.")
+
+    if interval_anchor and not interval:
+        exit_with_error("An anchor date can only be provided with an interval schedule")
+
+    try:
+        with open("deployment.yaml", "r") as f:
+            base_deploy = yaml.safe_load(f)
+    except FileNotFoundError:
+        app.console.print(
+            "No deployment.yaml file found, defaulting to empty values.", style="orange"
```

Comment:
I think this warning could be clearer.
```suggestion
            "No deployment.yaml file found, only provided CLI options will be used.", style="orange"
```

Reply:
Oh that's nice, thank you!